src/tests/param.cpp

```diff
@@ -14,43 +14,27 @@ REGISTER_TEST(ParamChange);
 
 ParamChange::ParamChange(const Context& context) : TestBase(context), _params_raw(context.system) {}
 
-TestBase::Result ParamChange::run()
+void ParamChange::run()
 {
     cout << "Setting param " << _config.name << " to " << _config.set_value << endl;
     ParamsRaw::Result set_result1 = _params_raw.set_param_int(_config.name, _config.set_value);
-    if (set_result1 != ParamsRaw::Result::SUCCESS) {
-        return Result::Failed;
-    }
+    EXPECT_EQ(set_result1, ParamsRaw::Result::SUCCESS);
 
     cout << "Checking param " << _config.name << " is " << _config.set_value << endl;
     const std::pair<ParamsRaw::Result, int> get_result1 = _params_raw.get_param_int(_config.name);
 
-    if (get_result1.first != ParamsRaw::Result::SUCCESS) {
-        return Result::Failed;
-    }
-
-    if (get_result1.second != _config.set_value) {
-        return Result::Failed;
-    }
+    EXPECT_EQ(get_result1.first, ParamsRaw::Result::SUCCESS);
+    EXPECT_EQ(get_result1.second, _config.set_value);
 
     cout << "Setting param " << _config.name << " to " << _config.reset_value << endl;
     ParamsRaw::Result set_result2 = _params_raw.set_param_int(_config.name, _config.reset_value);
-    if (set_result2 != ParamsRaw::Result::SUCCESS) {
-        return Result::Failed;
-    }
+    EXPECT_EQ(set_result2, ParamsRaw::Result::SUCCESS);
```

Comment:
Similar here: if setting the parameter fails, it's very likely that a read will fail as well (at least the value comparison).

Reply:
Which is fine I think.

Reply:
Exactly, that's what I was thinking.